ek9000App/src/devEK9000.cpp

```diff
@@ -770,13 +783,24 @@ int devEK9000::WriteWritelockMode(uint16_t mode) {
 }
 
 /* Read terminal ID */
-int devEK9000::ReadTerminalID(uint16_t termid, uint16_t& out) {
-	uint16_t tmp = 0;
-	this->doModbusIO(0, MODBUS_READ_INPUT_REGISTERS, 0x6000 + (termid), &tmp, 1);
-	if (tmp == 0)
-		return EK_ENOCONN;
-	out = tmp;
-	return EK_EOK;
+uint16_t devEK9000::ReadTerminalID(uint16_t index) {
+	assert(index < 0xFF);
+	if (m_readTerminals)
+		return m_terminals[index];
+
+	memset(m_terminals, 0, sizeof(m_terminals));
+	for (int off = 0; off < 255; off += 125) {
+		if (off > 0 && m_terminals[off] == 0)
```

Comment:
Trying to understand the logic of this loop...
So in iterations 2 and 3, the final element was written by the previous iteration (1 and 2) of the loop, and the final element should have something in it? What is that final element representing?

Reply:
Input register address 0x6000 to 0x60FF contain a numerical ID for each terminal. 0x6000 will always contain `9000` for ek9000. If the next terminal is `EL4004`, `0x6001` contains `4004` (in decimal), and so on. The rest of the registers are set to 0, so if the last element read is `0`, there are no more terminals on the rail and we can stop reading.

Reply:
Gotcha. Seems worthwhile to toss in a comment to that effect 👍